BSP/Simulator/Src/BSP_Fans.c

```diff
@@ -0,0 +1,69 @@
+#include "BSP_Fans.h"
+#include "simulator_conf.h"
+#include <unistd.h>
+#include <sys/file.h>
+#define MAX_SPEED    8
+#define NUM_FANS     4
+#define ARRAY_SIZE  (NUM_FANS * 3)
+
+static const char* file = GET_CSV_PATH(FANS_CSV_FILE);
+static char Fans[ARRAY_SIZE];
+
+/**
+ * @brief   Initialize all the GPIO pins connected to each fan
+ * @param   None
+ * @return  None
+ */
+void BSP_Fans_Init(void)
+{
+    // Creates file if none exists
+    FILE* fp = fopen(file, "w+");
+    int fno = fileno(fp);
+    flock(fno, LOCK_EX);
+    fprintf(fp, "00,00,00,00"); 
+    flock(fno, LOCK_UN);
+    fclose(fp);
+}
+
+/**
+ * @brief   Sets fan duty cycle
+ * @param   dutyCycle: int for duty cycle amount in range 0-8
+ *          fan: fan number whose speed should be changed (1-4)
+ * @return  None
+ */
+ErrorStatus BSP_Fans_Set(uint8_t fan, uint32_t dutyCycle){
+    //if input is outside of bounds
+    if (fan < 1 || fan > NUM_FANS) return ERROR;
+    if (dutyCycle < 0 || dutyCycle > MAX_SPEED) return ERROR;
+    FILE* fp = fopen(file, "r"); //Open file for reading
+    int fno = fileno(fp);  //Lock
+    flock(fno, LOCK_EX);    
+    fgets(Fans, ARRAY_SIZE, fp); //read from file
+    flock(fno, LOCK_UN); //Unlock
+    fclose(fp); //Close file
+    fp = fopen(file, "w"); //Open file for writing
+    //convert integer to string and store in Fans array
+    //NOTE: WHEN WE ADD DOUBLE DIGIT SPEED SETTINGS, REMOVE THE ZERO IN THE SPRINTF CALL!!!!!

```

Comment:
If I have to place commas in between the number, I think it has to be a character data type. So when I read the data into an array, it has to be %c. That's why I wrote 00,00... so it could store data with two digits.

Reply:
Never mind, it worked